wrapanapi/lenovo.py

```diff
@@ -357,5 +360,84 @@ def inventory(self, *requested_items, **kwargs):
         requested_items = requested_items or self._inventory_available
         return {item: self._inventory_available[item](self, requester) for item in requested_items}
 
+    def get_network_devices(self, server_name):
+        addin_cards = self.get_addin_cards(server_name) or []
+        pci_devices = self.get_pci_devices(server_name) or []
+        network_devices = []
+
+        for addin_card in addin_cards:
+            if (self.is_network_device(addin_card) and not
+                    self.is_device_in_list(addin_card, network_devices)):
+                network_devices.append(addin_card)
+
+        for pci_device in pci_devices:
+            if (self.is_network_device(pci_device) and not
+                    self.is_device_in_list(pci_device, network_devices)):
+                network_devices.append(pci_device)
+
+        return network_devices
+
+    def get_storage_devices(self, server_name):
+        addin_cards = self.get_addin_cards(server_name) or []
+        pci_devices = self.get_pci_devices(server_name) or []
+        storage_devices = []
+
+        for addin_card in addin_cards:
+            if (self.is_storage_device(addin_card) and not
+                    self.is_device_in_list(addin_card, storage_devices)):
+                storage_devices.append(addin_card)
+
+        for pci_device in pci_devices:
+            if (self.is_storage_device(pci_device) and not
+                    self.is_device_in_list(pci_device, storage_devices)):
+                storage_devices.append(pci_device)
+
+        return storage_devices
+
+    def is_device_in_list(self, device, device_list):
+        device_id = self.get_device_unique_id(device)
+
+        for d in device_list:
+            if device_id == self.get_device_unique_id(d):
+                return True
+
+        return False
+
+    def is_network_device(self, device):
+        device_name = device.get("productName") or device.get("name")
+        device_name = device_name.lower()
+
+        return (device.get("class") == "Network controller" or
+                "nic" in device_name or
+                "ethernet" in device_name)
+
+    def is_storage_device(self, device):
+        device_name = device.get("productName") or device.get("name")
+        device_name = device_name.lower()
+
+        return (device.get("class") == "Mass storage controller" or
+                "serveraid" in device_name or
+                "sd media raid" in device_name)
+
+    def get_addin_cards(self, server_name):
+        server = self.get_server(server_name)
+
+        return server.get("addinCards")
+
+    def get_pci_devices(self, server_name):
+        server = self.get_server(server_name)
+
+        return server.get("pciDevices")
+
+    def get_device_unique_id(self, device):
+        unique_id = None
+
+        if device.get("uuid") is None:
```

Comment:
**Optional**

Looks like this can be squashed a bit to:
```
unique_id = device.get("uuid") or  device.get("pciBusNumber") + device.get("pciDeviceNumber")
```

Reply:
Actually now that I say that, the logic here looks backwards. If the `device['uuid']` is None, then we would create one by combining the pci bus/device numbers.

Reply:
If one of device.get("pciBusNumber") or device.get("pciDeviceNumber") is None you could hit a concatenation error too :D Might want to do something like:

`unique_id = device.get("uuid") or '{}{}'.format(device.get("pciBusNumber"), device.get("pciDeviceNumber"))`